### PR TITLE
Feature/37 login authentication logic

### PR DESCRIPTION
**Pull request by LordAldi**

### CODE REVIEW

File: src/providers/auth/context.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import mocks from "../../mocks";
+const AuthContext = createContext(null);
+
+export function AuthProvider({ children }) {
+  const [reqHeader, setReqHeader] = useState({
+    "Device-id": "deviceid",
+    Authorization: "",
+  });
+  const [user, setUser] = useState();
+  const [token, setToken] = useLocalStorageState("token", "");
```
Comment (ariefrahmansyah) [inline src/providers/auth/context.js]: Using `token` as local storage key is too generic. Please rename it to `auth-token`

File: src/components/header/Header.js [hunk 2]
```diff
@@ -48,7 +49,7 @@ const Header = () => {
           <Col xs={5} sm={9} md={8} lg={6}>
             <div className="profile">
               <Avatar size={48} icon={<UserOutlined />} id="avatar" />
-              <p>Fadel Muhammad</p>
+              <p>{user ? user.username : "login please"}</p>
```
Comment (ariefrahmansyah) [inline src/components/header/Header.js]: Kalo user gak login, dia gak bisa buka halaman products dan lain-lain kan, which is otomatis gak bisa akses header juga kan? Kalo gitu, gak usah dipake kondisi `login please` lagi. Kita assume kalo Header udah kebuka, orangnya udah login.

File: src/providers/auth/context.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      mock: mocks.login,
+    }
+  );
+
+  const memoedValue = useMemo(
+    () => ({
+      user,
+      login,
+      token,
+      getCurrentUser,
```
Comment (ariefrahmansyah) [inline src/providers/auth/context.js]: Why do we need `getCurrentUser` here?

Comment (ariefrahmansyah) [reply src/providers/auth/context.js]: Can we directly call Users API after login and save the result to user field?

File: src/components/header/Header.js (hunk 2 above)
Comment (LordAldi) [reply src/components/header/Header.js]: okee kak, ini cuma buat test berhasil atau idk dio login, cuma testing. 
nanti ak langsung implement privateroute di pr ini atau buat baru?

File: src/providers/auth/context.js (hunk 3 above)
Comment (LordAldi) [reply src/providers/auth/context.js]: if we refresh page we must get current user again without re-login

Comment (LordAldi) [reply src/providers/auth/context.js]: https://dev.to/finiam/predictable-react-authentication-with-the-context-api-g10
this is my reference for this implementation

Comment (LordAldi) [reply src/providers/auth/context.js]: we dont need that, my bad. i already make useeffect on this file using getCurrentUser

File: src/components/header/Header.js (hunk 2 above)
Comment (LordAldi) [reply src/components/header/Header.js]: tapi ad issue #44 , dan dk mungkin kayakny kak ecommerce harus daftar dlu untuk liat"

**Review by ariefrahmansyah — APPROVED** [review]
LGTM! Amazing work!

**Review by fadelmvi — APPROVED** [review]
LGTM

**Review by fadelarmando — APPROVED** [review]
lgtm

**Pull request merged**